Contribution

RossCattero opened the pull request:
For trivia system;

RossCattero committed:
Contribution

For trivia system;

RossCattero committed:
Edit;

Added AddPlayerOption support;

swampservers commented on lua/autorun/sh_netstream.lua:
```diff
@@ -0,0 +1,164 @@
+--[[
```
you don't need this shit or sh_pon, use built in networking

RossCattero replied:
Okie. sh_pon and netstream just makes code looks more compact, but as you wish.

swampservers commented on lua/autorun/client/cl_ross_derma.lua:
```diff
@@ -0,0 +1,152 @@
+end;
+
+// I've taken it from the helix;
+function BetterWrapText(text, maxWidth, font)
```
Use my draw.WrappedText https://github.com/swampservers/contrib/blob/20f52368d7e9135be53213934ff3b5b6f96f811e/lua/autorun/client/stylesheet.lua

Do not make your code larger than it needs to be!!!! excess/duplicate code is a maintenance nightmare

RossCattero replied:
Ok, I'll use yours instead.

Where is it larger than it should be?

swampservers commented on lua/autorun/client/cl_votestuff_init.lua:
```diff
@@ -0,0 +1,77 @@
+})
+
+surface.CreateFont( "Anwser-font", {
+	font = "Century Gothic",
```
is this font on people's PC?

RossCattero replied:
Yes, it is. It's the default font on every **windows** PCs. For linux or MAC it will use the default fonts for their system.

swampservers commented on lua/autorun/sh_votestuff.lua:
```diff
@@ -0,0 +1,25 @@
+// Shared utils for votestuff by Ross;
+
+function RFormatTime(time)
```
https://wiki.facepunch.com/gmod/string.FormattedTime

RossCattero replied:
> https://wiki.facepunch.com/gmod/string.FormattedTime

I've done this function before this appeared in gmod I think. I'll use it, thanks for point

swampservers commented on lua/autorun/client/cl_ross_derma.lua:
```diff
@@ -0,0 +1,152 @@
+    return lines
+end
+    
+function panelMeta:Close()
```
i would rather not put this in, it might cause issues and its totally out of scope of this change. 

(meaning, this whole file, should be removed, and i am just commenting here as an example of something that seems like a potential glitch)

RossCattero replied:
> i would rather not put this in, it might cause issues and its totally out of scope of this change.
> 
> (meaning, this whole file, should be removed, and i am just commenting here as an example of something that seems like a potential glitch)

Agreed. I left this file for future purposes if I'll make a new VGUIs in future. But I'm going to remove it

RossCattero committed:
Changes;

1. Remove netstream and pon shit; Now using the default nets.
2. Minimized the code a bit where it's affordable;
3. Removed old functions for time, used new from gmod;
4. Used text wrapping from the gamemode instead of mine;

swampservers commented on lua/autorun/server/sv_votestuff.lua:
```diff
@@ -0,0 +1,53 @@
+local uniqueID = 'FetchQuestions'
+local time = Seconds[math.random(1, #Seconds)]
+timer.Create(uniqueID, time, 0, function()
+	if !timer.Exists(uniqueID) then timer.Remove(uniqueID) return; end;
```
line is unused (and makes no sense, you're removing something you know doesn't exist)

swampservers commented on lua/autorun/client/cl_votestuff_init.lua:
```diff
@@ -0,0 +1,73 @@
+if SERVER then return end;
```
won't happen, why leave this check in?

swampservers commented on lua/autorun/client/cl_votestuff_init.lua:
```diff
@@ -0,0 +1,73 @@
+surface.CreateFont( "Question-font", {
+	font = "Century Gothic",
+	size = ScreenScale(7),
+	scanlines = 0,
```
generally you shouldn't specify parameters that match the default values, and certainly not for this

swampservers commented on lua/autorun/client/cl_votestuff_init.lua:
```diff
@@ -0,0 +1,73 @@
+})
+
+net.Receive("tv::SendTasks", function( len )
+		local time = net.ReadUInt(16);
```
You don't need to change it as I have a thing that will fix this automatically, but for future reference - you're not using standard lua syntax, you're using retarded Garry syntax, which is considered amateur

- don't put ; at the end of lines
- use "not" instead of !
- use "and" instead of &&
- use "or" instead of ||

swampservers commented on lua/autorun/client/cl_votestuff_init.lua:
```diff
@@ -0,0 +1,73 @@
+				return true;
+		end, 
+		function() 
+			local lFont = "Question-font"
```
these variables are frivolous just put the string into the function call

swampservers commented on lua/autorun/client/cl_votestuff_init.lua:
```diff
@@ -0,0 +1,73 @@
+	extended = true,
+})
+
+surface.CreateFont( "Anwser-font", {
```
misspelled

swampservers commented on lua/autorun/client/cl_votestuff_init.lua:
```diff
@@ -0,0 +1,73 @@
+if SERVER then return end;
+
+local filterPattern = "&#?[%w]+;"
```
this should be on line 45

swampservers replied:
i have literally no clue what this is for outside of its proper context

swampservers commented on lua/autorun/server/sv_votestuff.lua:
```diff
@@ -0,0 +1,53 @@
+		local correct = Curr_Task.correct == anwser;
+
+		if correct then
+				-- client:SS_GivePoints( GivePoints )
```
if this code was enabled, a hacker could just spam this net message and give himself infinite points... come on man

swampservers commented on lua/autorun/server/sv_votestuff.lua:
```diff
@@ -0,0 +1,53 @@
+end);
+
+net.Receive("tv::SendAnswer", function( len, ply )
+		if !ply:IsValid() || !Curr_Task.correct then return end;
```
ply will always be valid

swampservers commented on lua/autorun/server/sv_votestuff.lua:
```diff
@@ -0,0 +1,53 @@
+util.AddNetworkString( "tv::SendAnswer" )
+
+Seconds = { // Each X seconds will appear a vote for all online players. They will have X/2 seconds to anwser;
+	10,
```
So you have logic lower down for selecting a random number from this table, but there is only one value in the table. This code is obviously unfinished and i'm not interested in finishing it for you.

Also "Seconds" is an inappropriate name for a global variable; it is likely to conflict with something else with equivalently careless variable naming.

swampservers commented on lua/autorun/server/sv_votestuff.lua:
```diff
@@ -0,0 +1,53 @@
+Curr_Task = Curr_Task || {}
+
+// Creating a random timer on init;
+local uniqueID = 'FetchQuestions'
```
frivolous variable

swampservers commented on lua/autorun/server/sv_votestuff.lua:
```diff
@@ -0,0 +1,53 @@
+if !SERVER then return end
```
frivolous check

swampservers commented:
Sorry about the delay in reviewing this
